docs/learn/learn-staking-miner.md

```diff
@@ -0,0 +1,79 @@
+---
+id: learn-staking-miner
+title: Staking Miner
+sidebar_label: Staking Miner
+description: Learn about Phragmén optimization on-chain via staking miners
+keywords: [staking, npos, miner, phragmén]
+slug: ../learn-staking-miner
+---
+
+:::caution
+The staking-miner code is experimental and it is still in development phase. Use is at your own discression, some funds might be lost.
+:::
+
+Staking miner is a passive validator functionality that runs when a new set of validators need to be elected for NPoS. This is a computationally intense process, hence the mining. Staking miners compete with each other to produce election solutions which consist of a validator set, stake distribution across that set, and a score indicating how optimal the solution is. Stakign miners run the sequential Phragmén algortihm to produce results, and the result is then sent as a transaction to the relay chain via a normal signed extrinsic. The transaction requires a bond, and a transaction fee. The best solution is rewarded
+which in the least covers the transaction fee, and the bond is returned to the account. If the solution is not valid, the bond and fee are lost.
+
+Staking miner uses the election pallet called `pallet_election_provider_multi_phase` and can only produce solutions during the signed phase of the pallets life cycle. Once the signed phase is over and the unsigned phase starts, only the off-chain workers can provide election results.
+
+
+## NPoS election optimization
+
+A basic election solution is a simple distribution of stake across validators, but this can be optimized for better distribution equaling a higher security score. The staking miner does not act as a validator and focuses solely on the election result and optimization of the solution. It connects to a specified chain and keeps listening to new signed phase of the election pallet in order to submit solutions to the NPoS election. When the correct time comes, it computes its solution and submit it to the chain. The default miner algorithm is sequential Phragmén with a configurable number of balancing iterations that improve the score.
```

Comment:
```suggestion
A basic election solution is a simple distribution of stake across validators, but this can be optimized for better distribution equaling a higher security score. The staking miner does not act as a validator and focuses solely on the election result and optimization of the solution. It connects to a specified chain and keeps listening to new signed phase of the election pallet in order to submit solutions to the NPoS election. When the correct time comes, it computes its solution and submits it to the chain. The default miner algorithm is sequential Phragmén with a configurable number of balancing iterations that improve the score.
```